CI failed with error that is a bit weird, because it is related to SSL and vertx, but not to this particular PR in theory:

```
2020-06-15T21:43:56.7706037Z Caused by: java.lang.LinkageError: loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd attempted duplicate class definition for io.vertx.core.net.impl.SSLHelper. (io.vertx.core.net.impl.SSLHelper is in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd, parent loader 'app')
2020-06-15T21:43:56.7708527Z 	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
2020-06-15T21:43:56.7710770Z 	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
2020-06-15T21:43:56.7713268Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:79)
2020-06-15T21:43:56.7715686Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:44)
2020-06-15T21:43:56.7718259Z 	at io.vertx.core.http.impl.HttpServerImpl.<init>(HttpServerImpl.java:93)
2020-06-15T21:43:56.7720769Z 	at io.vertx.core.impl.VertxImpl.createHttpServer(VertxImpl.java:310)
2020-06-15T21:43:56.7723388Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:759)
2020-06-15T21:43:56.7725311Z 	at io.vertx.core.Verticle.start(Verticle.java:66)
2020-06-15T21:43:56.7727722Z 	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$9(DeploymentManager.java:556)
2020-06-15T21:43:56.7730263Z 	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)
2020-06-15T21:43:56.7733198Z 	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
2020-06-15T21:43:56.7736078Z 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
2020-06-15T21:43:56.7738859Z 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
2020-06-15T21:43:56.7741256Z 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
2020-06-15T21:43:56.7744281Z 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
2020-06-15T21:43:56.7747935Z 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2020-06-15T21:43:56.7750831Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2020-06-15T21:43:56.7753149Z 	at java.base/java.lang.Thread.run(Thread.java:834)
```

I've restarted it.

--- a/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java
+++ b/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java
@@ -34,7 +34,7 @@ public class ReactiveSessionProducer {
     @RequestScoped
     @DefaultBean
     public Uni<Mutiny.Session> mutinySession() {
-        return mutinySessionFactory.openSession();
+        return mutinySessionFactory.openSession().cache();
     }
 
     public void disposeStageSession(@Disposes CompletionStage<Stage.Session> reactiveSession) {
@@ -45,11 +45,7 @@ public class ReactiveSessionProducer {
     }
 
     public void disposeMutinySession(@Disposes Uni<Mutiny.Session> reactiveSession) {
-        // TODO: @AGG this is not working, need to check w/ Clement to figure out the proper way
-        reactiveSession.on().termination((session, ex, cancelled) -> {
-            if (session != null)
-                session.close();
-        });
+        reactiveSession.subscribe().with(Mutiny.Session::close);
     }
 
 }

--- a/integration-tests/hibernate-reactive-db2/src/main/java/io/quarkus/it/hibernate/reactive/db2/HibernateReactiveDB2TestEndpoint.java
+++ b/integration-tests/hibernate-reactive-db2/src/main/java/io/quarkus/it/hibernate/reactive/db2/HibernateReactiveDB2TestEndpoint.java
@@ -132,9 +132,8 @@ public class HibernateReactiveDB2TestEndpoint {
     }
 
     private Uni<RowSet<Row>> populateDB() {
-        return db2Pool.getConnection()
-                .flatMap(c -> c.preparedQuery("DELETE FROM Pig").execute().map(junk -> c))
-                .flatMap(c -> c.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
+        return db2Pool.query("DELETE FROM Pig").execute()
+                .flatMap(junk -> db2Pool.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
     }
 
     private Uni<String> selectNameFromId(Integer id) {

--- a/integration-tests/hibernate-reactive-mysql/src/main/java/io/quarkus/it/hibernate/reactive/mysql/HibernateReactiveMySQLTestEndpoint.java
+++ b/integration-tests/hibernate-reactive-mysql/src/main/java/io/quarkus/it/hibernate/reactive/mysql/HibernateReactiveMySQLTestEndpoint.java
@@ -132,9 +132,8 @@ public class HibernateReactiveMySQLTestEndpoint {
     }
 
     private Uni<RowSet<Row>> populateDB() {
-        return mysqlPool.getConnection()
-                .flatMap(c -> c.preparedQuery("DELETE FROM Pig").execute().map(junk -> c))
-                .flatMap(c -> c.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
+        return mysqlPool.query("DELETE FROM Pig").execute()
+                .flatMap(junk -> mysqlPool.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
     }
 
     private Uni<String> selectNameFromId(Integer id) {

--- a/integration-tests/hibernate-reactive/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpoint.java
+++ b/integration-tests/hibernate-reactive/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpoint.java
@@ -132,9 +132,8 @@ public class HibernateReactiveTestEndpoint {
     }
 
     private Uni<RowSet<Row>> populateDB() {
-        return pgPool.getConnection()
-                .flatMap(c -> c.preparedQuery("DELETE FROM Pig").execute().map(junk -> c))
-                .flatMap(c -> c.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
+        return pgPool.query("DELETE FROM Pig").execute()
+                .flatMap(junk -> pgPool.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
     }
 
     private Uni<String> selectNameFromId(Integer id) {

